src/classes/ERR_ExceptionHandler.cls

```diff
@@ -98,22 +98,30 @@ public with sharing class ERR_ExceptionHandler {
         return false;
     }
 
-    public static void handleAfflNullRecordTypeException(Affiliation__c affl, List<String> validRecordTypes) {
-        if (affl.Account__c != null) {
-            // Get related Account's Record Type Name from 'Affiliation_Type__c' field
-            String afflAccRecTypeName;
-            if (affl.Affiliation_Type__c != null) {
-                AFFL_MultiRecordTypeMapper affilMapper = new AFFL_MultiRecordTypeMapper();
-                afflAccRecTypeName = affilMapper.accRecTypeLabelToAPI.get(affl.Affiliation_Type__c);
-            }
-            //If parent account exists for affiliation and account has no record type
-            //display the error message
-            if ((affl.Affiliation_Type__c != null && !validRecordTypes.contains(affl.Affiliation_Type__c)) && 
-                (String.isNotBlank(afflAccRecTypeName) &&  !validRecordTypes.contains(afflAccRecTypeName))) {
-                throw new ERR_ExceptionHandler.AffAccountRecordTypeMappingsException(Label.afflAccoutMappingError);
-            } else if (affl.Affiliation_Type__c == null) {
-                throw new ERR_ExceptionHandler.AfflAccountNoRecordTypeException(Label.afflAccountNoRecordType);
-            }
+    /*******************************************************************************************************
+    * @description Throws an exception if an affiliation's type is null or its parent account's record type is not in the list of valid record types.
+    * @param affl The affiliation to test.
+    * @param validRecordTypes The list of record types to validate against.
+    * @param accRecTypeLabelToAPI The map of account record type labels to account record type api names
+    */
+    public static void handleAfflNullRecordTypeException(Affiliation__c affl, List<String> validRecordTypes, Map<String,String> accRecTypeLabelToAPI) {
```

Comment:
This should work but second look on this makes me feel this method is doing something that it shouldn't. Maybe we should let exception handler only handle "exceptions" while letting the tdtm handle the logic by moving this method to the tdtm class. That way we don't have to worry about caching since mapper is cached anyway. Eventually we will have affiliation service to isolate the service logic and trigger context logic.

Reply:
This is not in the place I would like it to be. Moving it felt like it was outside the scope of this fix, especially given that we plan to replace this code entirely within the coming months.